backend/s3/config.go

```diff
@@ -73,9 +48,25 @@ func ParseConfig(s string) (interface{}, error) {
 		if url.Scheme == "http" {
 			cfg.UseHTTP = true
 		}
-
-		cfg.Bucket = url.Path[1:]
+		path = strings.SplitN(url.Path[1:], "/", 2)
+	} else if strings.HasPrefix(s, "s3:") {
+	       s = s[3:]
```

Comment:
That works :-)

I also pulled out the creation of the config into a separate method.